examples/tv-casting-app/android/App/app/src/main/jni/com/matter/casting/core/CastingPlayer.java

```diff
@@ -63,39 +63,97 @@ public interface CastingPlayer {
   int hashCode();
 
   /**
-   * Verifies that a connection exists with this CastingPlayer, or triggers a new session request.
-   * If the CastingApp does not have the nodeId and fabricIndex of this CastingPlayer cached on
-   * disk, this will execute the user directed commissioning process.
-   *
+   * @brief Verifies that a connection exists with this CastingPlayer, or triggers a new
+   *     commissioning session request. If the CastingApp does not have the nodeId and fabricIndex
+   *     of this CastingPlayer cached on disk, this will execute the User Directed Commissioning
+   *     (UDC) process by sending an IdentificationDeclaration message to the Commissioner. For
+   *     certain UDC features, where a Commissioner reply is expected, this API needs to be followed
+   *     up with the continueConnecting() API defiend below. See the Matter UDC specification or
+   *     parameter class definitions for details on features not included in the description below.
+   * @param connectionCallbacks contains the onSuccess, onFailure and onCommissionerDeclaration
+   *     callbacks defiend in ConnectCallbacks.java.
+   *     <p>onSuccess (Required): The callback called when the connection is established
+   *     successfully.
+   *     <p>onFailure (Required): The callback called with MatterError when the connection is fails
+   *     to establish.
+   *     <p>onCommissionerDeclaration (Optional): The callback called when the Commissionee receives
+   *     a CommissionerDeclaration message from the Commissioner. This callback is needed to support
+   *     UDC features where a reply from the Commissioner is expected. It provides information
+   *     indicating the Commissioner’s pre-commissioning state.
+   *     <p>For example: During Commissioner-Generated passcode commissioning, the Commissioner
+   *     replies with a CommissionerDeclaration message with PasscodeDialogDisplayed and
+   *     CommissionerPasscode set to true. Given these Commissioner state details, the client is
+   *     expected to perform some actions, detailed in the continueConnecting() API below, and then
+   *     call the continueConnecting() API to complete the process.
```

Comment:
What lines specifically? This is a copy form the documentation in CastingPlayer.h which you asked for in a previous PR: "In the docs above for VerifyOrEstablishConnection, can you add a mention of how ContinueConnecting is related to the former i.e. in what cases / on what callbacks may ContinueConnecting need to be called?"

Reply:
Yes, we should mention how ContinueConnecting and VerifyOrEstablishConnection are related and we can have the example at line 83 state what callback the client should expect, but the documentation for each of the individual callbacks (lines 74-82) should just exist in the ConnectionCallbacks class (so that it is easier to maintain in one place). Like this:

```suggestion
   * @param connectionCallbacks contains the onSuccess, onFailure and onCommissionerDeclaration
   *     callbacks defiend in ConnectionCallbacks.java.
   *     <p>For example: During Commissioner-Generated passcode commissioning, the Commissioner
   *     replies with a CommissionerDeclaration message with PasscodeDialogDisplayed and
   *     CommissionerPasscode set to true. Given these Commissioner state details, the client is
   *     expected to perform some actions on a call to connectionCallbacks.onCommissionerDeclaration(), detailed in the continueConnecting() API below, and then
   *     call the continueConnecting() API to complete the process.
```

Reply:
Sounds good.